standard_revisions/match_variables_from_two_versions_of_a_dataset.py

```diff
@@ -0,0 +1,313 @@
+"""Match variable IDs from and old version of a dataset to the analogous variables in the new version of the dataset.
+
+After a dataset has been uploaded to OWID's MySQL database, we need to pair new variable IDs with the old ones,
+so that all graphs update properly. If the variable names are identical, the task is trivial: find indexes of old
+variables and map them to the indexes of the identical variables in the new dataset.
+However, if variable names have changed (or the number of variables have changed) the pairing may need to be done
+manually. This script is a CLI tool that may help in either scenario.
+
+"""
+
+import argparse
+import json
+import os
+
+import pandas as pd
+from fuzzywuzzy import fuzz
+
+from db import get_connection
+
+CURRENT_DIR = os.path.dirname(__file__)
+OUTPUT_FILE = os.path.join(CURRENT_DIR, 'config', 'variable_replacements.json')
+# True to skip variables that are identical in old and new datasets, when running comparison.
+# If so, identical variables will be matched automatically.
+# False to include variables with identical names in comparison.
+OMIT_IDENTICAL = True
+# Function to use to match old and new variables.
+# MATCHING_FUNCTION = fuzz.partial_token_set_ratio
+MATCHING_FUNCTION = fuzz.partial_ratio
```

Comment:
Is there a reason `partial_ratio` is the default option? Might be nice for this to be determined via a `--scorer` command-line arg, since the right scorer may depend a lot on the specific naming conventions of the datasets at hand.

Reply:
Done